cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -307,6 +307,17 @@ template <typename T = uint8_t>
   return total_pages;
 }
 
+constexpr bool is_supported_encoding(Encoding enc)
```

Comment:
Similar to my comment above, can you add more clarification about the supported encoding? Like, why these encodings are supported while the others aren't, and where to look up?

Reply:
@ttnghia should I also add a comment in `setupLocalPageInfo` to modify this function if additional encodings are added?

Reply:
Yes, please do so. Ideally, the two functions should share code to always sync with each other but I don't know how to do that in the meantime. So please add a comment to remind other dev to sync them whenever they work on them.

Reply:
BTW, if that code (page_data.cu:L1172) throws error when an unsupported encoding is detected, why do you still need this PR?

Reply:
You mentioned that:
> the call to decode page data silently fails

So can we just modify that "silently fails" behavior to "loudly fails"? By doing so, we can avoid sync between different places.

Reply:
It actually doesn't throw an error. It just sets an error condition that is never checked.  I tried adding a CUDF_UNREACHABLE to the kernel, but that didn't halt execution as I thought it would.

Reply:
> You mentioned that:
> 
> > the call to decode page data silently fails
> 
> 
> 
> So can we just modify that "silently fails" behavior to "loudly fails"? By doing so, we can avoid sync between different places.

Yes, if I could actually get it to fail loud enough I'd prefer that approach.

Reply:
Thanks for that. Then let's move ahead with this PR while we may be able to improve the error check in the kernels later.
Sound like there is room for improvement @nvdbaranec.

Reply:
Do I need to use CUDF_CUDA_TRY?

Reply:
Throwing exception in device code is difficult. Typically, people store a global error state in global memory, which will be returned to the host after kernel launch. The caller will then investigate that error state and throw exception if needed.

Reply:
Ok, I added a comment to `setupLocalPageInfo`.